src/main/java/de/retest/recheck/review/ignore/ElementFilter.java

```diff
@@ -26,7 +26,7 @@ public boolean matches( final Element element ) {
 
 	@Override
 	public boolean matches( final Element element, final AttributeDifference attributeDifference ) {
-		return false;
+		return matcher.test( element );
```

Comment:
That was made intentionally and that is why we added both methods `matches(Element)` and `matches(Element, AttributeDifference)`. The task of a filter is to describe the matching based on the available data. An `ElementFilter` is only capable of ignoring elements, not attribute difference, since it has not that much information present. Analog the `ElementAttributeFilter` is not capable of filtering `matches(Element)` since it needs more information than there information is.

Thus the filter is not responsible for attribute filters to check for the element. The element in this method should only be used for context and not for main purpose (thats what `match(Element` is for). The calling implementation, however, need to verify that both methods are called in **every** circumstance.

Let's discuss this tomorrow.